Mathlib/MeasureTheory/Measure/FiniteMeasureExt.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2024 Jakob Stiefel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob Stiefel
+-/
+import Mathlib.Analysis.SpecialFunctions.MulExpNegSq
+import Mathlib.MeasureTheory.Measure.FiniteMeasure
+
+/-!
+# Extensionality of finite measures
+The main Result is `ext_of_forall_mem_subalgebra_integral_eq`:
+Let `A` be a StarSubalgebra of `C(E, 𝕜)` that separates points and whose elements are bounded. If
+the integrals of all elements `A` with respect to two finite measures `P, P'`coincide, then the
+measures coincide. In other words: If a Subalgebra separates points, it separates finite measures.
+-/
+
+open MeasureTheory Filter
+
+variable {E 𝕜: Type*} [RCLike 𝕜]
+    [MeasurableSpace E] [PseudoEMetricSpace E] [BorelSpace E] [CompleteSpace E]
+    [SecondCountableTopology E] {P P' : FiniteMeasure E}
```

Comment:
Instead of `[PseudoEMetricSpace E] [CompleteSpace E] [SecondCountableTopology E]`, can you assume `[TopologicalSpace E] [PolishSpace E]` which doesn't fix a distance? Since the result below does not involve the distance, it is probably a better formulation.

I see this relies on other results of the PR, but the same goes probably with the other results in the PR, all the way up to the results on the regularity of the measure.

Reply:
This can be done, but we lose some generality then (some metric space structure instead of pseudoemetric space). Would you nevertheless prefer this formulation?

Reply:
Do you have an application in mind which is not separated? Typically, probability theory is done with polish spaces (often without a metric, so the current version would not be applicable), but if you have a use for the non-separated situation we could consider introducing a pseudoPolish class or whatever.

Reply:
There are situations in probability where non-separated spaces appear, one usually modifies the space or metric then. I don't have a specific application in mind where this leads to a problem. Anyway, how about we state the theorem in the more general setting as it is now, and add the theorem on Polish spaces as an easy corollary (two line proof)? I just added this result